commit/gasworker/observation.go

```diff
@@ -0,0 +1,15 @@
+package gasworker
+
+import (
+	"context"
+)
+
+func (p *GasWorker) Observation(
+	ctx context.Context,
```

Comment:
Lets make this match libocr as much as possible for consistency. Needs to take in previous outcome (outctx) at least and maybe `types.Query`